ui-modules/blueprint-composer/app/components/util/d3-blueprint.js

```diff
@@ -838,11 +833,104 @@ export function D3Blueprint(container, options) {
                 return `M ${sourceNode.x},${sourceY} A ${dr},${dr} 0 0,${sweep} ${m.x},${m.y}`;
             });
 
-        relationData.exit()
+        relationArcs.exit()
             .transition()
             .duration(_configHolder.transition)
             .attr('opacity', 0)
             .remove();
+
+        // ====== RELATIONSHIP LABELS =========
+        // Draw relationship labels that follow paths, somewhere in the middle of paths.
+        // NOTE <textPath/> DECREASES THE UI PERFORMANCE, USE LABELS WITH CAUTION.
+
+        _relationLabels.selectAll('.' + LABEL_SELECTOR).remove(); // Re-draw labels every time, required to refresh changes.
+
+        // Group unique labels per path.
+        let labelsPerPath = {};
+        _d3DataHolder.visible.relationships.forEach(d => {
+            const key = getPathId(d);
+            if (!labelsPerPath[key]) {
+                labelsPerPath[key] = new Set();
+            }
+            labelsPerPath[key].add(d.label);
+        });
+
+        const getLabelId = (d) => (getPathId(d) + '-' + d.label);
+        const MAX_LABELS_TO_DISPLAY = 2;
+
+        // Data of unique labels with info about other labels at the same path.
+        let labelsData = _d3DataHolder.visible.relationships.reduce((accumulator, d) => {
+            const pathKey = getPathId(d);
+            const labelKey = getLabelId(d);
+            const labelsCollectedForPath = Object.keys(accumulator).filter(k => k.startsWith(pathKey)).length;
+            if (labelsCollectedForPath <= MAX_LABELS_TO_DISPLAY && !accumulator[labelKey]) {
+                accumulator[labelKey] = d;
+                accumulator[labelKey].labels = Array.from(labelsPerPath[pathKey]); // path labels.
+                accumulator[labelKey].labelIndex = labelsCollectedForPath; // label index at path.
+            }
+            return accumulator;
+        }, {});
+
+        const CANVAS_COLOR = '#f5f6fa';
+        const OFFSET_RATIO_PX = -12;
```

Comment:
Same as above

Reply:
Addressed in 94485d2c28c48a6aadb92f167fddc284eab02e8f.